swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift

```diff
@@ -131,26 +131,25 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
   ) {
     Log.log("stopTunnel: Reason: \(reason)")
 
+    do {
+      // There's no good way to send data like this from the
+      // Network Extension to the GUI, so save it to a file for the GUI to read upon
+      // either status change or the next launch.
+      try String(reason.rawValue).write(
+        to: SharedAccess.providerStopReasonURL, atomically: true, encoding: .utf8)
+    } catch {
+      Log.error(
+        SharedAccess.Error.unableToWriteToFile(
+          SharedAccess.providerStopReasonURL,
+          error
+        )
+      )
+    }
+
     if case .authenticationCanceled = reason {
-      Task {
```

Comment:
This Task was removed because there are no async/await functions called from within it, so wrapping this with `Task` would just block the executor thread unless we decide to introduce `withCheckedContinuation`.